vendor/github.com/meroxa/meroxa-go/pkg/meroxa/options.go

```diff
@@ -90,3 +90,14 @@ func WithAccountUUID(accountUUID string) Option {
 		return nil
 	}
 }
+
+// WithReferer sets the Referer header to use for requests.
+func WithReferer(referer string) Option {
+	return func(r *Requester) error {
+		if r.headers == nil {
+			r.headers = make(http.Header)
+		}
+		r.headers["Referer"] = []string{referer}
```

Comment:
The Referer header always expects a URL; we shouldn't use it for something else.

Reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer